modules/Header/ui/CategoriesNav/CategoriesNavDesktop.tsx

```diff
@@ -127,23 +132,31 @@ export function CategoriesNavDesktop({
                                 <div className={styles.backdrop} />
                             </div>
                         </Transition>
-                        {open && <BlockPageScroll />}
+                        {open && <BlockPageScroll scrollBarWidth={scrollBarWidthRef.current} />}
                     </>
                 )}
             </Popover>
         </li>
     );
 }
 
-function BlockPageScroll() {
+function BlockPageScroll({ scrollBarWidth }: BlockPageScroll.Props) {
```

Comment:
It would be nice to refactor it to a reusable hook called `useLockPageScroll(enabled: boolean)`.
I see no reason to have it as a component.

Reply:
Also, is there a reason for `scrollBarWidth` being passed from outside? 

The code related to `scrollBarWidth` seems to be possible to move into this component (or hook, once the component is converted to a hook).

Reply:
Converting it to a hook is not possible under the current scenario because:
- There is a `popover` component from `headless UI react` library that depends on the `open` and `close` value it exposes.
- Hook makes it unnecessarily complicated and requires us to add state at the top level that in turn is not compatible with the popover component.